# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/XPCJSRuntime.cpp
# Commit: ba8bd50a3a28
# Full Hash: ba8bd50a3a286518bd34857a6ea1ec457633648d
# Author: Denis Palmeiro <dpalmeiro@mozilla.com>
# Date: 2025-01-15 21:57:20
# Regressor Bug: 1932686
# File Overlap Count: 1
# Description:
#   Bug 1932686: Migrate legacy MemoryDistribution, Percentage, and Integer JS metrics to glean using a custom distribution.  r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231644
# ==============================================================================

diff -r 48aaa9a98d33 -r ba8bd50a3a28 js/xpconnect/src/XPCJSRuntime.cpp
--- a/js/xpconnect/src/XPCJSRuntime.cpp	Wed Jan 15 15:24:03 2025 +0000
+++ b/js/xpconnect/src/XPCJSRuntime.cpp	Wed Jan 15 15:30:15 2025 +0000
@@ -2689,6 +2689,51 @@
       glean::javascript_gc::task_start_delay.AccumulateRawDuration(
           TimeDuration::FromMicroseconds(sample));
       break;
+    case JSMetric::GC_MMU_50:
+      glean::javascript_gc::mmu_50.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_NURSERY_PROMOTION_RATE:
+      glean::javascript_gc::nursery_promotion_rate.AccumulateSingleSample(
+          sample);
+      break;
+    case JSMetric::GC_TENURED_SURVIVAL_RATE:
+      glean::javascript_gc::tenured_survival_rate.AccumulateSingleSample(
+          sample);
+      break;
+    case JSMetric::GC_PARALLEL_MARK_UTILIZATION:
+      glean::javascript_gc::parallel_mark_utilization.AccumulateSingleSample(
+          sample);
+      break;
+    case JSMetric::GC_NURSERY_BYTES_2:
+      glean::javascript_gc::nursery_bytes.Accumulate(sample);
+      break;
+    case JSMetric::GC_EFFECTIVENESS:
+      glean::javascript_gc::effectiveness.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_ZONE_COUNT:
+      glean::javascript_gc::zone_count.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_ZONES_COLLECTED:
+      glean::javascript_gc::zones_collected.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_PRETENURE_COUNT_2:
+      glean::javascript_gc::pretenure_count.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_MARK_RATE_2:
+      glean::javascript_gc::mark_rate.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_SLICE_COUNT:
+      glean::javascript_gc::slice_count.AccumulateSingleSample(sample);
+      break;
+    case JSMetric::GC_PARALLEL_MARK_SPEEDUP:
+      glean::javascript_gc::parallel_mark_speedup.AccumulateSingleSample(
+          sample);
+      break;
+    case JSMetric::GC_PARALLEL_MARK_INTERRUPTIONS:
+      glean::javascript_gc::parallel_mark_interruptions.AccumulateSingleSample(
+          sample);
+      break;
+
     default:
       // The rest aren't relayed to Glean.
       break;
